tests/system/source/main.d

```diff
@@ -51,13 +54,55 @@ private immutable Address[] Addrs = [
     { host: "127.0.0.1", port: 4002, },
 ];
 
+/// keep retrying 'check' condition until it's troe or the timer expires.
+/// If the timer expires, throw an exception. See also agora.test.Base.retryFor
+private void retryFor (lazy bool check, Duration timeout,
```

Comment:
Also this is mostly copy-pasted code. We should ideally just move it to a common module which the integration test can import, as well as the unittest modules. Would that be ok? Or do we just go with copy-pasting for now? (I prefer the former).

Reply:
https://github.com/bpfkorea/agora/blob/0120d6832a6112338d8588480400beb240ea60be/source/agora/utils/Test.d#L3-L19